andersoncd/penalties.py

```diff
@@ -22,13 +23,13 @@ def prox_1d(self, value, stepsize, j):
         """Proximal operator of penalty for feature j."""
 
     @abstractmethod
-    def subdiff_distance(self, w, neg_grad, ws):
+    def subdiff_distance(self, w, grad, ws):
         """Distance of gradient to subdifferential of penalty for feature j.
 
         w : array, shape (n_features,)
             Coefficient vector.
-        neg_grad: array, shape (n_features,)
-            Minus the value of the gradient of the datafit at w.
+        grad: array, shape (n_features,)
```

Comment:
with the code of the L1 penalty, it seems that grad has the size of the working set